project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js

```diff
@@ -77,11 +77,14 @@ export default class HeatMap extends React.Component {
   componentWillReceiveProps(nextProps) {
     const { placeid, defaultZoom } = this.state;
     const { targetBbox } = this.props;
-
-    if (hasChanged(this.props, nextProps) && nextProps.selectedplace.placeid && placeid !== nextProps.selectedplace.placeid) {
+    const { dashboardIsLoadedFromShareLink } = nextProps;
+    
+    if (hasChanged(this.props, nextProps) && dashboardIsLoadedFromShareLink && !this.state.sharedLinkMapRepositions) {
```

Comment:
Please extract the `hasChanged(this.props, nextProps))` into a variable since it's a potentially expensive deep equality check and there is on need here to re-compute it 3 times.